service-provider.md

```diff
@@ -3,4 +3,116 @@ description: Service Bindings for Service Providers
 permalink: /service-provider/
 ---
 
-# TODO
+Service providers expose bindings through a Secret resource with data required for connectivity.  The specification's [Provisioned Service section][provisioned-service] describes this in detail.  Alternatively, the specification also support [Direct Secret Reference][direct-secret-reference].  The only important part required for Direct Secret Reference is a Secret resource with data required for connectivity.  But if you are creating a Provisioned Service (preferred approach), you also need a custom resource with `.status.binding.name` attribute pointing to the Secret resource.
+
+Here is an example custom resource:
+
+```
+apiVersion: exmple.dev/v1beta1
+kind: Database
+metadata:
+  name: database-service
+...
+status:
+  binding:
+    name: production-db-secret
+```
+
+In the above example, `production-db-secret` is the name of the Secret resource with data entries required for connectivity.  The Secret resource should contain a `type` entry that can be used for identifying the service.  It helps the application to indentify the service as a relational database, key-value store, or a cache server.  There is no standardization on the value for type, but you can see some good examples in the Spring Cloud Bindings. Few examples:
+
+- cassandra
+- couchbase
+- db2
+- elasticsearch
+- kafka
+- ldap
+- mongodb
+- mysql
+- neo4j
+- oracle
+- postgresql
+- rabbitmq
+- redis
+- sqlserver
+- vault
+
+The type of the Secret resource should have a special value based on the `type` entry value.  The format is `servicebinding.io/{type}`.  For example, if the type entry is `mysql` the Secret's type value should be `servicebinding.io/mysql`.  This recommendation helps to query Secret resources of particular type using field-selector. For example:
```

Comment:
*thought*: It feels like the use of `type` here is really confusing because, depending on where you are in a given sentence, you either mean a type key in the payload or the type key in the spec.  It might be worthwhile being a bit more explicit about which you mean in every usage.

Reply:
I tried to address this issue by mentioning "field" when referring to the type of resource. And use "data" entry value for data value.
fe5638987aedd460d44e3d6e339fc7fc0faf25ce